backword-compatibility/class-adst-plugins-stats-api-bcp.php

```diff
@@ -0,0 +1,631 @@
+		}
+	}
+	/**
+	 * Human Readable Format
+	 *
+	 * @param int $plugin_count Get Count of plugin.
+	 * @return float $plugin_count Get human readable format.
+	 */
+	public function bsf_display_human_readable( $plugin_count ) {
+		$plugin_count = ( 0 + str_replace( ',', '', $plugin_count ) );
+		if ( ! is_numeric( $plugin_count ) ) {
+			return false;
+		}
+		$choice = get_option( 'adst_info' );
+		if ( 'K' === $choice['Rchoice'] ) {
+			return round( ( $plugin_count / 1000 ), 2 ) . $choice['Field1'];
+		} elseif ( 'M' === $choice['Rchoice'] ) {
+			return round( ( $plugin_count / 1000000 ), 4 ) . $choice['Field2'];
+		} elseif ( 'normal' === $choice['Rchoice'] ) {
+			return number_format( $plugin_count, 0, '', $choice['Symbol'] );
+		}
+		return $plugin_count;
+	}
+	/**
+	 * Shortcode
+	 *
+	 * @param int $atts Get attributes plugin Slug.
+	 * @return array $plugin Get plugin Details.
+	 */
+	public function display_plugin__version( $atts ) {
+		$wp_plugin_slug = $this->get_plugin_shortcode_slug( $atts );
+		if ( '' !== $wp_plugin_slug ) {
+			$api_params = array(
+				'plugin'   => $wp_plugin_slug,
+				'per_page' => self::$per_page,
+				'fields'   => array(
+					'homepage'        => false,
+					'description'     => false,
+					'screenshot_url'  => false,
+					'active_installs' => true,
+				),
+			);
+
+			$plugin = $this->get_api_data( 'plugin_information', $api_params );
+
+			if ( ! empty( $plugin ) ) {
+					return $plugin['version'];
+			} else {
+				return __( 'Plugin data is empty!', 'wp-themes-plugins-stats' );
+			}
+		}
+	}
+	/**
+	 * Shortcode
+	 *
+	 * @param int $atts Get attributes plugin Slug.
+	 * @return array $plugin Get plugin Details.
+	 */
+	public function display_plugin__ratings( $atts ) {
+		$wp_plugin_slug = $this->get_plugin_shortcode_slug( $atts );
+		if ( '' !== $wp_plugin_slug ) {
+			$api_params = array(
+				'plugin'   => $wp_plugin_slug,
+				'per_page' => self::$per_page,
+				'fields'   => array(
+					'homepage'       => false,
+					'description'    => false,
+					'screenshot_url' => false,
+					'num_ratings'    => true,
+				),
+			);
+
+			$plugin = $this->get_api_data( 'plugin_information', $api_params );
+
+			if ( ! empty( $plugin ) ) {
+					return $plugin['num_ratings'];
+			} else {
+				return __( 'Plugin data is empty!', 'wp-themes-plugins-stats' );
+			}
+		}
+	}
+	/**
+	 * Shortcode
+	 *
+	 * @param int $atts Get attributes plugin Slug.
+	 * @return array $plugin Get plugin Details.
+	 */
+	public function display_plugin__five_star_ratings( $atts ) {
+		$wp_plugin_slug = $this->get_plugin_shortcode_slug( $atts );
+		if ( '' !== $wp_plugin_slug ) {
+			$api_params = array(
+				'plugin'   => $wp_plugin_slug,
+				'per_page' => self::$per_page,
+				'fields'   => array(
+					'homepage'       => false,
+					'description'    => false,
+					'screenshot_url' => false,
+					'rating'         => true,
+				),
+			);
+
+			$plugin = $this->get_api_data( 'plugin_information', $api_params );
```

Comment:
This block of code is repeated in display_plugin__ratings function also. Can we make a common helper function for this?